streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStreamThreadTest.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.test.TestUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class GlobalStreamThreadTest {
+    private final KStreamBuilder builder = new KStreamBuilder();
+    private final MockConsumer<byte[], byte[]> mockConsumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+    private GlobalStreamThread globalStreamThread;
+
+    @Before
+    public void before() {
+        builder.globalTable("foo", "bar");
+        final HashMap<String, Object> properties = new HashMap<>();
+        properties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "blah");
+        properties.put(StreamsConfig.APPLICATION_ID_CONFIG, "blah");
+        final StreamsConfig config = new StreamsConfig(properties);
+        globalStreamThread = new GlobalStreamThread(builder.buildGlobalStateTopology(),
+                                                    config,
+                                                    mockConsumer,
+                                                    new StateDirectory("appId", TestUtils.tempDirectory().getPath()),
+                                                    new Metrics(),
+                                                    new MockTime(),
+                                                    "client");
+    }
+
+    @Test
+    public void shouldThrowStreamsExceptionOnStartupIfThereIsAnException() throws Exception {
+        // should throw as the MockConsumer hasn't been configured and there are no
+        // partitions available
+        try {
+            globalStreamThread.start();
+            fail("Should have thrown StreamsException if start up failed");
+        } catch (StreamsException e) {
+            // ok
+        }
+        assertFalse(globalStreamThread.stillRunning());
+    }
+
+
+    @Test
+    public void shouldBeRunningAfterSuccesulStart() throws Exception {
+        initializeConsumer();
+        globalStreamThread.start();
+        assertTrue(globalStreamThread.stillRunning());
+    }
+
+    @Test
+    public void shouldStopRunningWhenClosedByUser() throws Exception {
+        initializeConsumer();
+        globalStreamThread.start();
+        globalStreamThread.close();
+        globalStreamThread.join();
+    }
+
+    @Test
+    public void shouldCloseStateStoresOnClose() throws Exception {
```

Comment:
Nope - they are testing different things that happen on close. I prefer it this way as it is easier to just read the test names to see what should be happening rather than have to read through the assertions.